File: utilcode/src/test/java/com/blankj/utilcode/util/StringUtilsTest.java
package com.blankj.utilcode.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertTrue;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/08/16
 *     desc  : StringUtils 单元测试
 * </pre>
 */
public class StringUtilsTest {

    @Test
    public void isEmpty() throws Exception {
        assertTrue(StringUtils.Companion.isEmpty(""));
        assertTrue(StringUtils.Companion.isEmpty(null));
        assertFalse(StringUtils.Companion.isEmpty(" "));
    }

    @Test
    public void isSpace() throws Exception {
        assertTrue(StringUtils.Companion.isSpace(""));
        assertTrue(StringUtils.Companion.isSpace(null));
        assertTrue(StringUtils.Companion.isSpace(" "));
        assertTrue(StringUtils.Companion.isSpace("　 \n\t\r"));
    }

    @Test
    public void equals() throws Exception {
        assertTrue(StringUtils.Companion.equals(null, null));
        assertTrue(StringUtils.Companion.equals("blankj", "blankj"));
        assertFalse(StringUtils.Companion.equals("blankj", "Blankj"));
    }

    @Test
    public void equalsIgnoreCase() throws Exception {
        assertTrue(StringUtils.Companion.equalsIgnoreCase(null, null));
        assertFalse(StringUtils.Companion.equalsIgnoreCase(null, "blankj"));
        assertTrue(StringUtils.Companion.equalsIgnoreCase("blankj", "Blankj"));
        assertTrue(StringUtils.Companion.equalsIgnoreCase("blankj", "blankj"));
        assertFalse(StringUtils.Companion.equalsIgnoreCase("blankj", "blank"));
    }

    @Test
    public void null2Length0() throws Exception {
        assertEquals("", StringUtils.Companion.null2Length0(null));
    }

    @Test
    public void length() throws Exception {
        assertEquals(0, StringUtils.Companion.length(null));
        assertEquals(0, StringUtils.Companion.length(""));
        assertEquals(6, StringUtils.Companion.length("blankj"));
    }

    @Test
    public void upperFirstLetter() throws Exception {
        assertEquals("Blankj", StringUtils.Companion.upperFirstLetter("blankj"));
        assertEquals("Blankj", StringUtils.Companion.upperFirstLetter("Blankj"));
        assertEquals("1Blankj", StringUtils.Companion.upperFirstLetter("1Blankj"));
    }

    @Test
    public void lowerFirstLetter() throws Exception {
        assertEquals("blankj", StringUtils.Companion.lowerFirstLetter("blankj"));
        assertEquals("blankj", StringUtils.Companion.lowerFirstLetter("Blankj"));
        assertEquals("1blankj", StringUtils.Companion.lowerFirstLetter("1blankj"));
    }

    @Test
    public void reverse() throws Exception {
        assertEquals("jknalb", StringUtils.Companion.reverse("blankj"));
        assertEquals("knalb", StringUtils.Companion.reverse("blank"));
        assertEquals("文中试测", StringUtils.Companion.reverse("测试中文"));
        assertNull(StringUtils.Companion.reverse(null));
    }

    @Test
    public void toDBC() throws Exception {
        assertEquals(" ,.&", StringUtils.Companion.toDBC("　，．＆"));
    }

    @Test
    public void toSBC() throws Exception {
        assertEquals("　，．＆", StringUtils.Companion.toSBC(" ,.&"));
    }
}

File: utilcode/src/test/java/com/blankj/utilcode/util/StringUtilsTest.kt
package com.blankj.utilcode.util

import org.junit.Assert.*
import org.junit.Test

/**
 * <pre>
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/08/16
 * desc  : StringUtils 单元测试
</pre> *
 */
class StringUtilsTest {

    @Test
    @Throws(Exception::class)
    fun isEmpty() {
        assertTrue(StringUtils.isEmpty(""))
        assertTrue(StringUtils.isEmpty(null))
        assertFalse(StringUtils.isEmpty(" "))
    }

    @Test
    @Throws(Exception::class)
    fun isSpace() {
        assertTrue(StringUtils.isSpace(""))
        assertTrue(StringUtils.isSpace(null))
        assertTrue(StringUtils.isSpace(" "))
        assertTrue(StringUtils.isSpace("　 \n\t\r"))
    }

    @Test
    @Throws(Exception::class)
    fun equals() {
        assertTrue(StringUtils.equals(null, null))
        assertTrue(StringUtils.equals("blankj", "blankj"))
        assertFalse(StringUtils.equals("blankj", "Blankj"))
    }

    @Test
    @Throws(Exception::class)
    fun equalsIgnoreCase() {
        assertTrue(StringUtils.equalsIgnoreCase(null, null))
        assertFalse(StringUtils.equalsIgnoreCase(null, "blankj"))
        assertTrue(StringUtils.equalsIgnoreCase("blankj", "Blankj"))
        assertTrue(StringUtils.equalsIgnoreCase("blankj", "blankj"))
        assertFalse(StringUtils.equalsIgnoreCase("blankj", "blank"))
    }

    @Test
    @Throws(Exception::class)
    fun null2Length0() {
        assertEquals("", StringUtils.null2Length0(null))
    }

    @Test
    @Throws(Exception::class)
    fun length() {
        assertEquals(0, StringUtils.length(null).toLong())
        assertEquals(0, StringUtils.length("").toLong())
        assertEquals(6, StringUtils.length("blankj").toLong())
    }

    @Test
    @Throws(Exception::class)
    fun upperFirstLetter() {
        assertEquals("Blankj", StringUtils.upperFirstLetter("blankj"))
        assertEquals("Blankj", StringUtils.upperFirstLetter("Blankj"))
        assertEquals("1Blankj", StringUtils.upperFirstLetter("1Blankj"))
    }

    @Test
    @Throws(Exception::class)
    fun lowerFirstLetter() {
        assertEquals("blankj", StringUtils.lowerFirstLetter("blankj"))
        assertEquals("blankj", StringUtils.lowerFirstLetter("Blankj"))
        assertEquals("1blankj", StringUtils.lowerFirstLetter("1blankj"))
    }

    @Test
    @Throws(Exception::class)
    fun reverse() {
        assertEquals("jknalb", StringUtils.reverse("blankj"))
        assertEquals("knalb", StringUtils.reverse("blank"))
        assertEquals("文中试测", StringUtils.reverse("测试中文"))
        assertNull(StringUtils.reverse(null!!))
    }

    @Test
    @Throws(Exception::class)
    fun toDBC() {
        assertEquals(" ,.&", StringUtils.toDBC("　，．＆"))
    }

    @Test
    @Throws(Exception::class)
    fun toSBC() {
        assertEquals("　，．＆", StringUtils.toSBC(" ,.&"))
    }
}